### ISSUE TITLE
Multi-Levels - Level selection show all objects of others levels

### ISSUE THREAD

**Issue by vpnobregasmarthome** [adizanni/floor3d-card#109]
Hello, i have the project of my house with several floors and I exported it through the available plugin. 
When adding this card, I can see the various floors on the left side but when I select only a lower floor, the objects on the upper floor are visible. Any idea what could be causing this?

**Comment by adizanni** [comment]
The functionality is quite new so it may have some bugs. Normally the level buttons on the top left corner are acting as indipendent switches, if you click on a level you hide/show the level objects and if you click on the top icon you make them all show. The only way for me to test is if you share your sh3d model.

**Comment by vpnobregasmarthome** [comment]
[Projeto_v13.zip](https://github.com/adizanni/floor3d-card/files/8651761/Projeto_v13.zip)
![Screenshot_16](https://user-images.githubusercontent.com/95915015/167413211-ca20a5b0-e166-4c8c-b75b-2363885f174f.png)

thanks for the quick response. I attached the project and an image of the floor 0 selection and the visible objects of the upper floors

**Comment by adizanni** [comment]
Hello @vpnobregasmarthome, to see the association between an object and the level you need to ceck the furniture list below the furniture catalogue in the left part of the sweethome3d screen. For wall and other core elements, the only way is to switch between the tab of the models. Look at this picture:

![image](https://user-images.githubusercontent.com/35622920/167482562-a7b81b2e-deda-46f0-9fb9-b23b4137189c.png)

**Comment by vpnobregasmarthome** [comment]
I confirmed in the objects properties and the level is correctly associated. From the image you can see that several objects are on different levels. Can you validate if it's a bug in the addon or in the way to validate the object level?
![Screenshot_17](https://user-images.githubusercontent.com/95915015/167587026-becc159c-e618-4e01-b550-a2b7748106b4.png)

**Comment by adizanni** [comment]
Would you be ok to share your sh3d model and send me via cloud drive ?

**Comment by adizanni** [comment]
Ok, I've found the bug. In order to be compliant with the .obj waterfront format, I only accept standard characters (a-z A-Z 0..9). It is quite a limitation (chinese guys already complained in a previous issue). Before I address this issue, please try to comply with an object naming that does not contain special portuguese chracters. I may try to disable the waterfront standard in a future release but with unpredictable results........

**Comment by vpnobregasmarthome** [comment]
> Ok, I've found the bug. In order to be compliant with the .obj waterfront format, I only accept standard characters (a-z A-Z 0..9). It is quite a limitation (chinese guys already complained in a previous issue). Before I address this issue, please try to comply with an object naming that does not contain special portuguese characters (berco televisao, etc.). I may try to disable the waterfront standard in a future release but with unpredictable results........ By the way your model is impressive and simple enough to be managed easily by the card. Very good.

thanks. I'll put the program in English and export the file again to test.

**Comment by adizanni** [comment]
I do not think you are forced to install the english version (as anyway the core elements like walls and rooms are not localized), but you need to rename all your objects. 
Also to claim this is not an issue but works as expected you can try to use the Export to obj function of Sweethome3d and you will see that special character are not well managed....

**Comment by vpnobregasmarthome** [comment]
I updated the object names but when I click on an object, a popup appears with the object's old name. When I'm mapping a light to objects, the name that appeared in the popup doesn't appear in the object list. This list already has the new name. Any idea what it could be? I have already cleared the browser cache.

**Comment by adizanni** [comment]
From what you describe, it seems that either you did not overwrite the old .obj file (only the json file) or the browser is picking the old .obj version. On top of clearing the cache (that should solve the problem), please try to version the name of the .obj file by adding a URL version (ex. home.obj?ver=1, and the same for the mtl file and for the json file). Try to do a visual  modification to your model to see if the card loaded the new version.

**Comment by vpnobregasmarthome** [comment]
I changed the file name and I can already see the updated object name. Now I have another problem. I have a light and I want to turn on the light and see the effect of the light on on the card but I see that the state of the light entity changes when I click on the light but visually nothing changes.
it will be a problem with the configuration of the card?

`entities:
  - entity: light.controller_rgb_80ad05
    type3d: light
    object_id: Luz_Escritorio_2
    light:
      lumens: '800'
      shadow: 'yes'`

![Screenshot_18](https://user-images.githubusercontent.com/95915015/168281156-f1dc837e-7458-40bb-b78a-885c8d1a39e4.png)

**Comment by adizanni** [comment]
When you have solid lights like the one you show in the picture, try to use the vertical alignment = bottom. Because otherwise the light is stopped by the lamp itself.

**Comment by vpnobregasmarthome** [comment]
> by the la

I tried it but the light effect did not appear

**Comment by adizanni** [comment]
There is some light appearing in the picture, if it is not really the light effect; I think there is something wrong with the light entity or with the object itself: try to add the light to another object in the model (ex. the shelf) and see if it works. In the meantime I will do some test with the model you sent me.

**Comment by vpnobregasmarthome** [comment]
testing the part of turning the lights on/off, it doesn't work in chrome via laptop but in chrome via android I can see the light and the color associated with the led strip. It's just some cuts on the wall, which is flat, I don't understand that having these cuts in the light effect. Do you understand why this happens? shouldn't it work in chrome via laptop?
![Screenshot_19](https://user-images.githubusercontent.com/95915015/168563020-51c8270f-662a-4836-ae76-4405ab23c93d.png)

**Comment by adizanni** [comment]
The cuts are due to the shadow. I've not done enough test of lights in multilevel. It is always good to put ceilings in the room in sweethome, they will be transparent from above but they will not let the light out

**Comment by vpnobregasmarthome** [comment]
> 

I removed the shadow effect and the lines with the shadows on the walls no longer appear

**Comment by vpnobregasmarthome** [comment]
I always have ceilings, the point is that the ceiling gives rise to another level. To be able to see the lamps, I have to select the view without ceilings because then I can't select the lights. is there an option to choose a level by default instead of having them all selected? I ask this because whenever I look at the card, I have to take several levels to see the level I want. if there was a configuration parameter, it would open the card at the desired level.

**Comment by adizanni** [comment]
I can easily put another button to unselect all and then you select the level you want (it would be 2 clicks but no config to update). Otherwise I can have the same approach as Sweethome3d: Clicking on level 0 it shows only level 0, clicking on level 1 it shows 0+1, etc. 
If you do not mind I can turn this issue into an enhancement request. If you have some bugs related to how the lights are working in multi-level please post another one

**Comment by vpnobregasmarthome** [comment]
Following the sweet home approach was ideal. Select 0, 0 appears, select 1, 0+1 appears. 
Also a default way instead of showing all levels. 
Will you be able to develop this for a future release?

**Comment by V1per1331** [comment]
Hello Andy,
This card really works wonders, the only concern that bothers me, the level buttons do not appear on my card. The option to hide the level menus is of course off in the config.
However, I imported my project with your plugin, and also transformed them into '''.glb' with nodejs, everything went smoothly.
except that in summary I can not change levels
sweetHome version 7
exportHA up to date
could you make a tutorial or give me the procedure to follow please
Cordially
![image](https://user-images.githubusercontent.com/18025790/216475524-ac3fd67d-bb41-41aa-bc19-d55615c7a395.png)

**Comment by adizanni** [comment]
@V1per1331 apologies, I did not get the notification for this new post on the issue. if you use .obj and .mtl from the plugin does it work ?

**Comment by V1per1331** [comment]
Good morning,
thank you for your reply.
no even with .obj and .mtl it does not work at all.
I tested different configurations, I even came to do 2 exports, one for the ground floor and one for the floor, which I positioned in another dashboard, but I no longer have a solution.

Cordially

**Comment by adizanni** [comment]
In this case I need to test your model in Sweethome3d, I have no other clue.

**Comment by V1per1331** [comment]
ok how can i send it to you

**Comment by V1per1331** [comment]
https://www.swisstransfer.com/d/2aabf3b8-0194-4878-8e05-f40574c2572a

**Comment by MarkuZP** [comment]
Hello Andy, first of all. This is a cool card an feature. It would be awesome if you would could some work on it to improve it. My problem is also with the levels. I followed the hint with the the bounding to levels. That solved some issues but I have still a little random view of furniture and walls hinding even if they are bounded to the correct level. Sometimes it works, sometimes not. Browsercache is also cleared before.  I followed also the hint with the "groups into groups " failure and deleted this grouping. I could send you a link for Video here. And my GLB File too. My Home Floorplan is quiete big and detailed. Is there a chance that the Model is not reloading the whole project when I switch to another Browsertab?  Best regards Markuz Thank you!!

**Comment by MarkuZP** [comment]
<img width="1133" height="785" alt="Image" src="https://github.com/user-attachments/assets/ec649500-8418-409e-bd53-c505014b7cd7" />
